src/evict/evict_lru.c

```diff
@@ -2054,13 +2056,17 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
 
         /*
          * If the global transaction state hasn't changed since the last time we tried eviction,
-         * it's unlikely we can make progress. Similarly, if the most recent update on the page is
-         * not yet globally visible, eviction will fail. This heuristic avoids repeated attempts to
-         * evict the same page.
+         * it's unlikely we can make progress. Similarly, if the most recent update on the page from
+         * the transaction that is greater than the last running transaction, eviction will fail.
```

Comment:
This is not correct with durable history. May be state that this is something legacy.

Reply:
Fixed in f2722384d578b0a49f7374bdb2f6f349cdf71e3d